src/Raven.Server/Web/Operations/OperationsHandler.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+using Raven.Server.Documents;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Web.Operations
+{
+    public class OperationsHandler : DatabaseRequestHandler
+    {
+
+        [RavenAction("/databases/*/operations/next-operation-id", "GET")]
+        public Task GetNextOperationId()
```

Comment:
1) Yes, webscokets are probably the way to go. Note that we'll need to handle heartbeats here too. And yes, they have separate websockets pool.
2) No, we keep streaming status to the client as we go along in the operation.
3) No, the request continue working, but there is no report to the client (still can get the ops status, but probably won't).
4) We return the id as the first step in the process.
5) The connection will close, but we'll continue running the operation, just not report it.